File: windows/LanAudioBridge.Desktop/Core/Models/NetworkInfo.cs
namespace LanAudioBridge.Core;

/// <summary>
/// 网络状态信息快照
/// </summary>
public struct NetworkInfo
{
    /// <summary>是否已连接</summary>
    public bool IsConnected;

    /// <summary>当前活跃的传输类型</summary>
    public TransportType TransportType;

    /// <summary>网络质量</summary>
    public NetworkQuality Quality;

    /// <summary>WiFi SSID（如有）</summary>
    public string? Ssid;

    /// <summary>网络接口名</summary>
    public string? InterfaceName;
}

File: android/app/src/main/java/com/lanbridge/app/core/models/NetworkInfo.kt
package com.lanbridge.app.core.models

import com.lanbridge.app.core.enums.NetworkQuality
import com.lanbridge.app.core.enums.TransportType

/**
 * 网络状态信息快照
 */
data class NetworkInfo(
    /** 是否已连接 */
    val isConnected: Boolean,
    /** 当前活跃的传输类型 */
    val transportType: TransportType = TransportType.Udp,
    /** 网络质量 */
    val quality: NetworkQuality = NetworkQuality.Unknown,
    /** WiFi SSID（如有） */
    val ssid: String? = null,
    /** 网络接口名 */
    val interfaceName: String? = null
)
